fpa/interfaces.go

```diff
@@ -0,0 +1,52 @@
+package fpa
+
+import (
+	"context"
+	"io"
+
+	"github.com/ipfs/go-cid"
+)
+
+// WalletManager provides access to a Lotus wallet for a Lotus node.
+type WalletManager interface {
+	NewWallet(ctx context.Context, typ string) (string, error)
+	Balance(ctx context.Context, addr string) (uint64, error)
+}
+
+// Scheduler creates and manages Job which executes Cid configurations
+// in Hot and Cold layers, enables retrieval from those layers, and
+// allows watching for Job state changes.
+type Scheduler interface {
+	Enqueue(CidConfig) (JobID, error)
+	GetFromHot(ctx context.Context, c cid.Cid) (io.Reader, error)
+	GetJob(JobID) (Job, error)
+
+	Watch(InstanceID) <-chan Job
+	Unwatch(<-chan Job)
+}
+
+// HotLyer is a fast datastorage layer for storing and retrieving raw
+// data or Cids.
+type HotLayer interface {
```

Comment:
Wondering, maybe these should be `HotStorage` and `ColdStorage`? The word "layer" doesn't describe much about what it is.

Reply:
Yeah, I agree. _Layer_ sounds much more abstract of what this is describing. 
Will change this in the next PR 👍